Backport 2.28: Fix IAR warnings

paul-elliott-arm opened the pull request:
## Description

IAR was warning that conditional execution could bypass initialisation of variables, although those same variables were not used uninitialised, however this is a quick fix to keep IAR quiet.

This will need forward porting to development once it has been checked to be ok (I cannot run IAR here)

## Gatekeeper checklist

- [x] **changelog** provided
- [X] **forward-port** #7090 (See comment about #6754)
- [x] **tests** ~~provided, or~~ not required

tom-cosgrove-arm commented:
When you say "forward port done", are you referring to #6754 which includes the `exponent_bits_in_window` fix?

paul-elliott-arm commented:
Ah no, I meant it is yet to be done, and I hadn't yet spotted that, thank you.

daverodgman commented:
> should this have a `ChangeLog` entry? We do have entries such as `Fix compilation warnings with IAR toolchain`, and might be important to users of the library...

Agreed, but it should be integrated into the existing Changelog addition from #6888, rather than have separate bullet points for each warning fix.

Kasimashi commented:
Some warning are remaining please can you fix it before merging ? : 
#6924 

;)

paul-elliott-arm commented:
Added new requested fixes, and added an edit to the previous IAR changelog. @Kasimashi - please re-review, and could you also take a look at https://github.com/Mbed-TLS/mbedtls/pull/6754 and confirm that we do not need any of those warning fixes in 2.28.2

tom-cosgrove-arm commented on ChangeLog.d/fix-iar-warnings.txt:
```diff
@@ -1,2 +1,3 @@
 Bugfix
-   * Fix IAR compiler warnings. Contributed by Glenn Strauss in #3835.
+   * Fix IAR compiler warnings. Contributed by Glenn Strauss in #3835, and
+     reported by Kasimashi in #6924.
```
(minor) I think we normally just say

```
   * Fix IAR compiler warnings. Contributed by Glenn Strauss in #3835.
     Fixes #6924.
```

gilles-peskine-arm replied:
It's the other way round, actually: we do, in principle, credit external reporters in changelog entries. Always for security issues, not always for ordinary bugs. We generally don't credit contributors these days: it's understood that the credit is in the author fields in the git log.

tom-cosgrove-arm replied:
However, in this case the fix was only given in a comment on an issue, and so Glenn is not the author of the commit, and #6924 is not a security issue, but the reporting of warnings issued by a compiler. So yes "I think we normally just say" maybe should be "In this case I think we should just say"

gilles-peskine-arm reviewed (dismissed):
Approved — I don't know if this fixes all IAR warnings, but at least it's non-harmful.

gilles-peskine-arm commented:
The code style check is complaining.

Also, what's the status with forward-porting?

tom-cosgrove-arm reviewed (dismissed):
LGTM

paul-elliott-arm commented:
Rebased, and hopefully code-styled.

paul-elliott-arm commented:
Have provided my own forward port in order to get this one through (https://github.com/Mbed-TLS/mbedtls/pull/6754 contains much other work, which will also require backports, and this will muddy the water somewhat). I suggest we look at #6754 next.

tom-cosgrove-arm reviewed (approved):
LGTM

tom-cosgrove-arm commented:
CI is fully green

gilles-peskine-arm reviewed (approved):
I'm happy that the changes are non-harmful. I have not personally checked that they resolve the warnings as advertised, but I am satisfied that [the reporter has reported that they do](https://github.com/Mbed-TLS/mbedtls/issues/6924#issuecomment-1398791958).
